src/irc/core/irc-servers-setup.c

```diff
@@ -138,8 +138,6 @@ static void init_userinfo(void)
 	if (nick == NULL || *nick == '\0') {
 		str = g_getenv("IRCNICK");
 		settings_set_str("nick", str != NULL ? str : user_name);
-
-		nick = settings_get_str("nick");
```

Comment:
for example this statement here will resolve the precondition nick == NULL so that nick is guaranteed to be not NULL after this code block, in case of a future code addition which needs to make use of that variable